src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -294,3 +289,35 @@
                                                 {:event  :wallet/send-transaction
                                                  :error  error
                                                  :params request-params}))}]})))
+
+(rf/reg-event-fx
+ :navigation/wizard-send-flow
+ (fn [_ [{:keys [current-screen params is-first?]}]]
+   (when is-first?
+     (rf/dispatch [:wallet/clean-send-data]))
+   (rf/dispatch [:navigation/wizard
+                 {:params         params
+                  :current-screen current-screen
+                  :is-first?      is-first?
+                  :flow-config    wallet-events/send-asset-flow-config}])))
+
+(rf/reg-event-fx
+ :navigation/wizard-back-send-flow
+ (fn [{:keys [db]}]
+   (let [stack          (:modal-view-ids db)
+         current-screen (last stack)]
+     (case current-screen
```

Comment:
this is a bit strange to me, for the navigate forward we are passing the event? but for the navigate back we are putting them all in the navigate backwards.

maybe it's cleaner to just write new re-frame events for each and fire them from the given page?

Reply:
Maybe we are over-engineering here?

I'd say wizarding logic should only take care of navigation, not side effects of navigating.

Reply:
Yeah, I agree 👌